File: payola/s2js/runtime/client/src/main/scala/s2js/runtime/client/scala/util/control/Breaks.scala
/*                     __                                               *\
**     ________ ___   / /  ___     Scala API                            **
**    / __/ __// _ | / /  / _ |    (c) 2003-2011, LAMP/EPFL             **
**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
** /____/\___/_/ |_/____/_/ | |                                         **
**                          |/                                          **
\*                                                                      */

package s2js.runtime.client.scala.util.control

class BreakControlException extends Exception

object Breaks
{
    /**
      * A block from which one can exit with a `break`. The `break` may be
      * executed further down in the call stack provided that it is called on the
      * exact same instance of `Breaks`.
      */
    def breakable(op: () => Unit) {
        try {
            op()
        } catch {
            case _: BreakControlException =>
            case ex => throw ex
        }
    }

    def break() {
        throw new BreakControlException
    }
}

File: payola/web/server/public/javascripts/runtime/s2js/runtime/scala/util/control/Breaks.js
goog.provide('scala.util.control.BreakControlException');
goog.provide('scala.util.control.Breaks');
goog.require('scala.Exception');
scala.util.control.BreakControlException = function() {
var self = this;
goog.base(self);
};
goog.inherits(scala.util.control.BreakControlException, scala.Exception);
scala.util.control.BreakControlException.prototype.__class__ = new s2js.Class('scala.util.control.BreakControlException', [scala.Exception]);
scala.util.control.Breaks.breakable = function(op) {
var self = this;
try {
op();
} catch ($ex$1) {
(function() {
if (s2js.isInstanceOf($ex$1, 'scala.util.control.BreakControlException')) {

return;
}
if (true) {
var ex = $ex$1;
(function() {
throw ex;
})();
return;
}
throw $ex$1;
})();
}

};
scala.util.control.Breaks.$break = function() {
var self = this;
(function() {
throw new scala.util.control.BreakControlException();
})();
};
scala.util.control.Breaks.__class__ = new s2js.Class('scala.util.control.Breaks', []);
